flit/validate.py

```diff
@@ -54,10 +52,27 @@ def _download_classifiers():
     cache_dir = get_cache_dir()
     try:
         cache_dir.mkdir(parents=True)
-    except FileExistsError:
+    except (FileExistsError, PermissionError, OSError):
+        # readonly mounted file raises OSError
```

Comment:
I don't really like catching any OSError - that's quite a broad category. OSError objects normally have an `e.errno` attribute which you can check against constants in the [`errno` module](https://docs.python.org/3/library/errno.html) to look for specific errors. E.g. it sounds like this might be `EROFS` (read only file system) - but check if that's what you're hitting.

This pattern was more common in Python 2 code, before subclasses like `FileExistsError` and `PermissionError` were split out as their own subclasses.

Reply:
I was running into this problem on a Mac with 10.15 "Catalina" running Python 3.7. The root and other system directories are read-only. 

Trying to create the dir "/nonexistent/" raises a `OSError` for a readonly file system – unfortunately not a subclassed error like `PermissionError` but an `OSError` itself:

```
>>> open("/foo", "w")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 30] Read-only file system: '/foo'
```

I changed the try-except-block to be a little bit more verbose and check for an explicit `errno.EROFS`